File: src/main/java/fr/aiko/Ryoko/parser/Token.java
package fr.aiko.Ryoko.parser;

public class Token {
    private final int startPosition;
    public TokenType type;
    public String value;

    public Token(TokenType type, String value, int startPosition) {
        this.type = type;
        this.value = value;
        this.startPosition = startPosition;
    }

    public TokenType getType() {
        return type;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String s) {
        value = s;
    }

    public int getStart() {
        return startPosition;
    }
}

File: src/main/kotlin/studio/karllang/karl/lexer/Token.kt
package studio.karllang.karl.lexer

class Token(private val type: TokenType, private var value: String, private val position: Int, private val line: Int) {
    fun getType(): TokenType {
        return type
    }

    fun getValue(): String {
        return value
    }

    fun setValue(newValue: String) {
        value = newValue
    }

    fun getPosition(): Int {
        return position
    }

    fun getLine(): Int {
        return line
    }
}